app/javascript/packages/document-capture/components/in-person-full-address-search.tsx

```diff
@@ -141,17 +144,38 @@ function FullAddressSearch({
     validatedZipCodeFieldRef,
   } = useUspsLocations(locationsURL);
 
-  const inputChangeHandler =
+  const inputChangeHandlerForZipCode =
     <T extends HTMLElement & { value: string }>(input) =>
     (event: React.ChangeEvent<T>) => {
       const { target } = event;
-      input(target.value);
+      const scrubbedZip = target.value.replace(/[^0-9]/g, '');
+      if (scrubbedZip.length > 5) {
+        const zipcodePieces: string[] = [];
+        zipcodePieces.push(scrubbedZip.substring(0, 5));
+        zipcodePieces.push(scrubbedZip.substring(5));
+        input(zipcodePieces.join('-'));
+      } else {
+        input(scrubbedZip);
+      }
+    };
+
+  const inputChangeHandler =
+    <T extends HTMLElement & { value: string }>(input, ref) =>
+    (event: React.ChangeEvent<T>) => {
+      const { target } = event;
+      if (ref === 'address') {
+        input(target.value.replace(/[^a-zA-Z0-9_ ]/gi, '').trimStart());
+      } else if (ref === 'city') {
+        input(target.value.replace(/[^a-zA-Z0-9-'_ ]/gi, '').trimStart());
+      } else {
+        input(target.value.trimStart());
+      }
     };
 
-  const onAddressChange = inputChangeHandler(setAddressValue);
-  const onCityChange = inputChangeHandler(setCityValue);
-  const onStateChange = inputChangeHandler(setStateValue);
-  const onZipCodeChange = inputChangeHandler(setZipCodeValue);
+  const onAddressChange = inputChangeHandler(setAddressValue, 'address');
+  const onCityChange = inputChangeHandler(setCityValue, 'city');
+  const onStateChange = inputChangeHandler(setStateValue, 'state');
```

Comment:
If each of these has a custom `if` branch in `inputChangeHandler` to handle the how the value is formatted, is there much value in having a common function, vs. having a separate function for each?

e.g.

```suggestion
  type InputChangeEvent = React.ChangeEvent<HTMLInputElement>;
  const onAddressChange = (event: InputChangeEvent) => setAddressValue(target.value.replace(/[^a-zA-Z0-9_ ]/gi, '').trimStart());
  const onCityChange = (event: InputChangeEvent) => setCityValue(target.value.replace(/[^a-zA-Z0-9-'_ ]/gi, '').trimStart());
  const onStateChange = (event: InputChangeEvent) => setStateValue(target.value.trimStart());
```